executor/calibrate_resource.go

```diff
@@ -83,13 +91,77 @@ type baseResourceCost struct {
 	writeReqCount uint64
 }
 
+const lowUsageThreshold = 10.
+const percentOfPass = 0.9
+const discardRate = 0.1
+
 type calibrateResourceExec struct {
 	baseExecutor
-
+	optionList   []*ast.DynamicCalibrateResourceOption
 	workloadType ast.CalibrateResourceType
 	done         bool
 }
 
+func (e *calibrateResourceExec) checkDynamicCalibrateOptions() (string, string, error) {
+	var startTime, endTime time.Time
+	var startTimeStr, endTimeStr string
+	checkDurationFn := func() error {
+		// check the duration
+		duration := endTime.Sub(startTime)
+		if duration > time.Hour*24 {
+			return errors.Errorf("the duration of calibration is too long.")
+		}
+		if duration < time.Minute*10 {
+			return errors.Errorf("the duration of calibration is too short.")
+		}
+		return nil
+	}
+	if len(e.optionList) == 2 {
+		if e.optionList[0].Tp != ast.CalibrateStartTime || (e.optionList[1].Tp != ast.CalibrateEndTime && e.optionList[1].Tp != ast.CalibrateDuration) {
```

Comment:
`e.optionList[1].Tp != ast.CalibrateEndTime && e.optionList[1].Tp != ast.CalibrateDuration`
I'm not sure why the same parameter would have && judgment twice

Reply:
Because we're trying to determine if it's wrong. The expression to assert truth is `e.optionList[1].Tp == ast.CalibrateEndTime || e.optionList[1].Tp == ast.CalibrateDuration`